pymatgen/analysis/diffusion/aimd/pathway.py

```diff
@@ -237,21 +237,22 @@ def to_chgcar(self, filename="CHGCAR.vasp"):
             f.write(" 1.00 \n")
 
             for i in range(3):
-                f.write(" {0} {1} {2} \n".format(*self.structure.lattice.matrix[i, :]))
+                v = self.structure.lattice.matrix[i, :]
+                f.write(f" {v[0]} {v[1]} {v[2]} \n")
 
             f.write(" " + " ".join(symbols) + "\n")
             f.write(" " + " ".join(natoms) + "\n")
             f.write("direct\n")
             for fcoord in init_fcoords:
-                f.write(" {0:.8f}  {1:.8f}  {2:.8f} \n".format(*fcoord))
+                f.write(" {fcoord[0]:.8f}  {fcoord[1]:.8f}  {fcoord[2]:.8f} \n")
```

Comment:
I doubt these actually work. You need to write tests. Otherwise I am not accepting this PR.

Reply:
This is how I checked whether or not this would work. If you don't believe this will work, can you please elaborate on what you think is incorrect? Or better yet, contribute to correcting the error you have identified.

![image](https://user-images.githubusercontent.com/42151899/139917209-e49c9485-3a39-41fa-85f9-070f7ee8eead.png)

It is not reasonable to request someone else write tests for code they're not familiar with that is also unrelated to the original PR. My goal was to be helpful since I can see why fixing some string formatting is annoying and apparently a low priority and am sorry I made some errors. I will submit another commit to correct the f-string formatting shortly.

From my perspective, the original PR that bumped pylint from 2.10.2 to 2.11.1 that caused the pylint tests to fail (https://github.com/materialsvirtuallab/pymatgen-analysis-diffusion/pull/257) shouldn't have been accepted and it should be the repository's owner responsibility for correcting these types of issues. If you would prefer to exclude any of the commits in this PR related to pylint that's fine by me. But refusing to merge the PR because of an unreasonable request to add tests for code not related to the PR just discourages people from contributing to this code.

Reply:
Yes, I believe the subsequent f-string changes were due to some unrelated linting errors coming from the main branch, not due to the new and tested code in this PR. I would suggest reverting the f-string commits and leave them for the original author to resolve.

Reply:
Let's look at line 247. That is not an f-string. Literally, there is no f in front of the string. So when you have a statement like `f.write(" {fcoord[0]:.8f}  {fcoord[1]:.8f}  {fcoord[2]:.8f} \n")`, your code is literally writing `" {fcoord[0]:.8f}  {fcoord[1]:.8f}  {fcoord[2]:.8f} \n"` and not whatever value is in fcoord. Let me demonstrate it easily:

```
In [1]: with open("test.txt", "wt") as f:
   ...:     f.write(" {fcoord[0]:.8f}  {fcoord[1]:.8f}  {fcoord[2]:.8f} \n")
   ...:

In [2]: ! cat test.txt
 {fcoord[0]:.8f}  {fcoord[1]:.8f}  {fcoord[2]:.8f}
```

This is the literal output of that statement. 

The "check" that you wrote above is correct in that you actually have an f in front of the string. Now this is repeated throughout the entire code base.